### PR TITLE
Fix triple check when some parts are unknown

### PR DESCRIPTION
**Pull request by martong**

### CODE REVIEW

File: lib/Tooling/CrossTranslationUnit.cpp [hunk 1]
```diff
@@ -183,13 +183,19 @@ const FunctionDecl *CrossTranslationUnit::getCrossTUDefinition(
...
+      const auto& TripleFrom = Unit->getASTContext().getTargetInfo().getTriple();
       // The imported AST had been generated for a different target
-      if (Context.getTargetInfo().getTriple() !=
-          Unit->getASTContext().getTargetInfo().getTriple()) {
+      // TODO use equality operator. Note, for some unknown reason when we do
+      // in-memory/on-the-fly CTU (i.e when the compilation db is given) some
+      // parts of the triple in the loaded ASTContext can be unknown while the
+      // very same parts in the target ASTContext are known. Thus we check for
+      // the known parts only.
+      if (!TripleTo.hasEqualKnownFields(TripleFrom)) {
```
Comment (bruntib) [inline lib/Tooling/CrossTranslationUnit.cpp]: This function doesn't exist, thus the code doesn't compile.

Comment (martong) [reply lib/Tooling/CrossTranslationUnit.cpp]: Yeah :(
Actually Triple.h is in LLVM repo, we have to fork that into Ericsson as well :(

**Comment by Xazax-hun** [comment]
Can't we have a free function?

File: lib/Tooling/CrossTranslationUnit.cpp (hunk 1 above)
Comment (martong) [reply lib/Tooling/CrossTranslationUnit.cpp]: Looks like I can access the private variables with accessors, so we can create the function in the Clang repo, working on it.

Comment (dkrupp) [reply lib/Tooling/CrossTranslationUnit.cpp]: no, I think we should not patch LLVM just for this reason. Please implement this function into Clang.
I think we should revert this PR https://github.com/Ericsson/clang/pull/226

until this is properly solved. This blocks us now.

**Pull request merged**